File: Data Structures & Algorithms/longest-increasing-subsequence/submission-3.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        
        int n = nums.length;
        int[] tails = new int[n];
        int size = 1;
        tails[0] = nums[0];

        for(var i = 1; i < n; i++) {
            if (nums[i] > tails[size - 1]){
                System.out.print(" nums[i]="+nums[i]);
                tails[size] = nums[i];
                size++;
            } 
            if(nums[i] < tails[size - 1]) {
                for (var j = 0; j < size; j++) {
                    if(nums[i] == tails[j]) break;
                    if (nums[i] < tails[j]) {
                        tails[j] = nums[i];
                        break;
                    }
                }
            }
        }
        for (var a : tails) {
            System.out.print(" tail = "+a);
        }
        return size;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) <= 0: return 0

        nums.sort()
        max_l = 1

        curr = nums[0]
        curr_l = 1
        print(nums)
        for i, num in enumerate(nums, start = 1):
            if num == curr + 1:
                curr_l += 1
            elif num == curr:
                continue
            else:
               curr_l = 1

            max_l = max_l if max_l > curr_l else curr_l
            curr = num 

        return max_l    